src/lib/protocols/dhcp.c

```diff
@@ -180,7 +186,24 @@ void ndpi_search_dhcp_udp(struct ndpi_detection_module_struct *ndpi_struct,
 	      //	    while(j < len) { printf( "%c", name[j]); j++; }; printf("\n");
 #endif
             ndpi_hostname_sni_set(flow, name, len);
-	  }
+	        } else if(id == 15 /* Domain Name */) {
+            char *name = (char*)&dhcp->options[i+2];
+            int j = 0;
+	    
+            j = ndpi_min(len, sizeof(flow->protos.dhcp.domain_name)-1);
+            strncpy((char*)flow->protos.dhcp.domain_name, name, j);
+            flow->protos.dhcp.domain_name[j] = '\0';
+          } else if(id == 50) /* Requested IP */ {
+            memcpy(&flow->protos.dhcp.requested_ip, (char*)&dhcp->options[i+2], len); 
```

Comment:
the industry standard defined for dhcp says this field should be of length 4 and is unchanging so having a define may make sense rather than pulling the length

Reply:
for cases where a field can have a variable length pulling len is necessary imo but for static fields define macros may make more sense